intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -928,3 +936,303 @@ def validate_ip(value):
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/NTP-Version
+ntp_version = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port')
+    ],
+    'optional_fields': [
+        ('protocol.transport', 'protocol'),
+        ('source.reverse_dns', 'hostname'),
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('extra.', 'version', validate_to_none),
+        ('extra.', 'clk_wander', validate_to_none),
+        ('extra.', 'clock', validate_to_none),
+        ('extra.', 'error', validate_to_none),
+        ('extra.', 'frequency', validate_to_none),
+        ('extra.', 'jitter', validate_to_none),
+        ('extra.', 'leap', validate_to_none),
+        ('extra.', 'mintc', validate_to_none),
+        ('extra.', 'noise', validate_to_none),
+        ('extra.', 'offset', validate_to_none),
+        ('extra.', 'peer', validate_to_none),
+        ('extra.', 'phase', validate_to_none),
+        ('extra.', 'poll', validate_to_none),
+        ('extra.', 'precision', validate_to_none),
+        ('extra.', 'processor', validate_to_none),
+        ('extra.', 'refid', validate_to_none),
+        ('extra.', 'reftime', validate_to_none),
+        ('extra.', 'rootdelay', validate_to_none),
+        ('extra.', 'rootdispersion', validate_to_none),
+        ('extra.', 'stability', validate_to_none),
+        ('extra.', 'state', validate_to_none),
+        ('extra.', 'stratum', validate_to_none),
+        ('extra.', 'system', validate_to_none),
+        ('extra.', 'tai', validate_to_none),
+        ('extra.', 'tc', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.type': 'vulnerable service',
+        'classification.identifier': 'openntpversion',
+        'feed.code': 'shadowserver-openntpversion',
+        'protocol.application': 'ntp',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Sandbox-URL
+sandbox_url = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('malware.hash.md5', 'md5hash'),
+        ('event_description.url', 'url'),
+        ('extra.', 'user_agent', validate_to_none),
+        ('extra.', 'host', validate_to_none),
+        ('extra.', 'method', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.type': 'malware',
+        'feed.code': 'shadowserver-sandboxurl',
+        'classification.identifier': 'sandboxurl',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Spam-URL
+spam_url = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
```

Comment:
`ip` holds the host of possibly unwanted content (It's not malicous, we never know). I doubt this can be used at all, because we don't know anything about it.

If we blame `src` (which is correct according to the description: * If the "src" IP is one of their systems, this means that your email server was the one sending, routing, or forwarding the Spam messages.* But `src` is not our country, Spamhaus sends the data to us, because of the `ip`, the last hop which relayed the message.

So I think we could save 2 events. First, the host of the linked website and second the relay.